File: backend/src/main/scala/TextAudioIndex.scala
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

case class TextAudioIndex(id: Int, samplingRate: Int, textStartPos: Int, textEndPos: Int, audioStartPos: Double, audioEndPos: Double, speakerKey: Int, labeled: Int, transcriptFileId: Int) {
}

object TextAudioIndex {
  implicit val encoder = deriveEncoder[TextAudioIndex]
  implicit val decoder = deriveDecoder[TextAudioIndex]
}

File: frontend/src/app/models/textAudioIndex.ts
export class TextAudioIndex {
  id: number;
  samplingRate: number;
  textStartPos: number;
  textEndPos: number;
  audioStartPos: number;
  audioEndPos: number;
  speakerKey: number;
  labeled: number;
  transcriptFileId: number;

  constructor(id: number, samplingRate: number, textStartPos: number, textEndPos: number, audioStartPos: number, audioEndPos: number, speakerKey: number, labeled: number, transcriptFileId: number) {
    this.id = id;
    this.samplingRate = samplingRate;
    this.textStartPos = textStartPos;
    this.textEndPos = textEndPos;
    this.audioStartPos = audioStartPos;
    this.audioEndPos = audioEndPos;
    this.speakerKey = speakerKey;
    this.labeled = labeled;
    this.transcriptFileId = transcriptFileId;
  }
}
